src/utils/UserUtils.tsx

```diff
@@ -0,0 +1,55 @@
+import { FIRST_ALPHABET_REGEX, SPLITTING_WORDS_REGEX } from '../@config/constants'
+
+/**
+ * This is a helper function to help the wallet user see prefilled initials.
+ *
+ * @param namePart1 is first part of the username. Also known as firstName
+ * @param namePart2 is 2nd part of the username. Also known as lastName, surname or remaining part of the name etc.
+ *
+ * @return the initials selected by default
+ */
+export const getDefaultUserNameInitials = (namePart1?: string, namePart2?: string): string => {
```

Comment:
Named changed.

Reply:
i am not sure what is happening here, i would expect 1 function called getInitials that gets 1 string, which is the full name of the user. and returns 2 letters

Reply:
just a quick search. something like this but maybe a little cleaner:

unction getInitials(fullName) {
  const nameArray = fullName.trim().split(" "); // split the full name into an array of individual names
  const firstInitial = nameArray[0][0]; // get the first letter of the first name
  const lastInitial = nameArray[nameArray.length - 1][0]; // get the first letter of the last name
  return `${firstInitial}${lastInitial}`; // combine the two initials into a string and return it
}

// Example usage:
const fullName = "John Smith Jr.";
const initials = getInitials(fullName);
console.log(initials); // output: "JS"

Reply:
We do not have any rule for what can be in the first and the last name the only checks are that it should not be empty.

So what is happening here is we Just select 2 alphabets ... These function just provide a user default initials. Even when the user enters a 
FirstName: '.@Sam
LastName: __--Pretty

The current implementation will just find SP. That's why so much is happening in the code.

Reply:
That will fail miserably with non alphabet names. Please simply trimm everything.  Split by whitespace. Take the first and last word and uppercase their first letters

Reply:
@nklomp that is exactly what I am doing ... And it is definitely working ... It's working for even weired inputs like all special characters. 
It will return a string '?'. But it will always respond with a usable string.

Reply:
So why are the regexes in there then that match the alphabet?

Creating an avatar should not be concerned with any input validation. If a user can enter '#$% 123' the result should be #1 in the avatar

Reply:
After personally talking to Niels last week the @Brummos suggestion was opted.